packages/frontend/core/src/components/blocksuite/block-suite-editor/specs/custom/root-block.ts

```diff
@@ -141,6 +161,190 @@ function getThemeExtension(framework: FrameworkProvider) {
+      : { edgeless: icons.edgeless, page: icons.page };
+  }
+
+  class AffineDocDisplayMetaService
+    extends LifeCycleWatcher
+    implements DocDisplayMetaExtension
+  {
+    static override key = 'doc-display-meta';
+
+    readonly disposables: Disposable[] = [];
+
+    readonly iconMap = new WeakMap<DocRecord, Signal<TemplateResult>>();
+
+    readonly titleMap = new WeakMap<DocRecord, Signal<string>>();
+
+    static override setup(di: Container) {
+      super.setup(di);
+      di.override(DocDisplayMetaProvider, this, [StdIdentifier]);
+    }
+
+    dispose() {
+      while (this.disposables.length > 0) {
+        this.disposables.pop()?.dispose();
+      }
+    }
+
+    icon(
+      docId: string,
+      { params, title, referenced = false }: DocDisplayMetaParams = {}
+    ): Signal<TemplateResult> {
+      const doc = docsService.list.doc$(docId).value;
+
+      if (!doc) {
+        return signal(icons.deleted);
+      }
+
+      let icon = this.iconMap.get(doc);
+      if (!icon) {
+        const docIcons = getDocIcons(referenced);
+
+        icon = signal(
+          doc.primaryMode$.value === 'edgeless'
+            ? docIcons.edgeless
+            : docIcons.page
+        );
+
+        const icon$ = LiveData.computed(get => {
+          const journalDate = docDisplayMetaService.toDayjs(
+            get(journalService.journalDate$(docId))
+          );
+          if (journalDate) {
+            return iconBuilder(
+              docDisplayMetaService.getJournalIcon(journalDate, { type: 'lit' })
+            );
+          }
+
+          return get(doc.primaryMode$) === 'edgeless'
+            ? docIcons.edgeless
+            : docIcons.page;
+        });
+        const subscription = icon$.subscribe({
+          next: value => {
+            if (!icon) return;
+            icon.value = value;
+          },
+        });
+        const disposable = {
+          dispose: () => subscription.unsubscribe(),
+        };
+
+        this.disposables.push(disposable);
+        this.disposables.push(
+          this.std.collection.slots.docRemoved
+            .filter(docId => docId === doc.id)
+            .once(() => {
+              const index = this.disposables.findIndex(d => d === disposable);
+              if (index !== -1) {
+                this.disposables.splice(index, 1);
+                disposable.dispose();
+              }
+              this.iconMap.delete(doc);
+            })
+        );
+        this.iconMap.set(doc, icon);
+      }
+
+      return computed(() => {
+        if (title) {
+          return icons.aliased;
+        }
+
+        if (referenceToNode({ pageId: docId, params })) {
+          return icons.linkedBlock;
+        }
+
+        return icon.value;
+      });
+    }
+
+    title(docId: string): Signal<string> {
+      const doc = docsService.list.doc$(docId).value;
+      if (!doc) return signal(I18n['com.affine.notFoundPage.title']());
+
+      let title = this.titleMap.get(doc);
```

Comment:
use docDisplayMetaService

Reply:
docId and links are a `1:N` relationship